components/Chips/src/MDCChipView.m

```diff
@@ -629,7 +635,47 @@ - (void)layoutSubviews {
 }
 
 - (CGRect)contentRect {
-  return UIEdgeInsetsInsetRect(self.bounds, self.contentPadding);
+  CGRect contentRect = UIEdgeInsetsInsetRect(self.bounds, self.contentPadding);
+  UIControlContentHorizontalAlignment alignment = self.contentHorizontalAlignment;
+  if (alignment != UIControlContentHorizontalAlignmentCenter) {
+    return contentRect;
+  }
+
+  // Calculate the minimum width needed for all the content. If it's less than contentSize.width,
+  // then inset to center. If not, just return contentRect.
+  CGFloat neededContentWidth = 0.0f;
+  CGSize maxContentSize = contentRect.size;
+
+  // If there's an imageView, add it and its padding.
+  if (self.showImageView || self.showSelectedImageView) {
+    CGFloat maxImageWidth = 0.0f;
+    if (self.showImageView) {
+      maxImageWidth = [self sizeForImageView:self.imageView maxSize:maxContentSize].width;
+    }
+    if (self.showSelectedImageView) {
+      maxImageWidth = MAX(
+          maxImageWidth, [self sizeForImageView:self.selectedImageView maxSize:maxContentSize].width);
+    }
+    neededContentWidth += maxImageWidth + UIEdgeInsetsHorizontal(self.imagePadding);
+  }
+
+  // Always add the title and its padding.
+  neededContentWidth += [_titleLabel sizeThatFits:maxContentSize].width;
```

Comment:
Please use property access instead of ivars directly.

s/_titleLabel/self.titleLabel/

and others below

Reply:
There are 15 references to _titleLabel in this class, and only 3 to self.titleLabel. I'm going to go for consistency here.

Reply:
I think in this case, following best practices should trump consistency. I wouldn't mind switching the file over or just being consistent within the new code (preferring properties).

From Apple's guidelines:

> Although **it’s best practice for an object to access its own properties using accessor methods or dot syntax**, it’s possible to access the instance variable directly from any of the instance methods in a class implementation. ...
In general, you should use accessor methods or dot syntax for property access even if you’re accessing an object’s properties from within its own implementation, in which case you should use self... **The exception to this rule is when writing initialization, deallocation or custom accessor methods**, as described later in this section. (Emphasis added)
> &mdash; [Encapsulating Data (Accessed: June 6, 2018)](https://developer.apple.com/library/archive/documentation/Cocoa/Conceptual/ProgrammingWithObjectiveC/EncapsulatingData/EncapsulatingData.html)

Reply:
The style guide that MDC says to use is the Google one (https://github.com/google/styleguide/blob/gh-pages/objcguide.md), which currently does not favor one or the other. But it does say this:
> When the style guide allows multiple options it is preferable to pick one option over mixed usage of multiple options. Using one style consistently throughout a codebase lets engineers focus on other (more important) issues. Consistency also enables better automation because consistent code allows more efficient development and operation of tools that format or refactor code. In many cases, rules that are attributed to "Be Consistent" boil down to "Just pick one and stop worrying about it"; the potential value of allowing flexibility on these points is outweighed by the cost of having people argue over them.

Reply:
Understood and I'm aware of the consistency guidelines as well.  I agree that this shouldn't be debated, I was just hoping we could clean up the code a bit since it was being worked on.

Reply:
I understand your desire for it to get cleaned up, but as an example, while writing this code, I wasn't sure whether to use _titleLabel or self.titleLabel (and some for other ivars vs properties). I looked through the code and found more usages of the ivar, so I used that. I'd prefer to do the cleanup in one fell swoop as to not add to the confusion (i.e. more people like me, editing code here and being unsure which to use).